cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/cube/DefaultCube.java

```diff
@@ -80,6 +87,8 @@ public DefaultCube(int[] resolutions, FactTableSupplier factTableSupplier,
       resolutionToFactTable.put(resolution, factTableSupplier.get(resolution, 3600));
     }
     this.aggregationAliasMap = aggregationAliasMap;
+    this.executorService = Executors.newFixedThreadPool(resolutions.length,
```

Comment:
So this will never get shutdown? How many instances of `DefaultCube` are we creating?

Reply:
I showdown this in the close() method. But the `DefaultCube` is created if `MetricStore` is injected, so there can be multiple instances. Is there a better way to deal with this scenario?

Reply:
Changed to not use a fixed thread pool